custom/icds/management/commands/contact_phone_fix.py

```diff
@@ -0,0 +1,174 @@
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.log import with_progress_bar
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.form_processor.backends.sql.dbaccessors import iter_all_rows
+
+from dimagi.utils.chunked import chunked
+from casexml.apps.case.mock import CaseBlock
+
+DOMAIN = "icds-cas"
+CASE_TYPE = "person"
+
+PHONE_NUMBER_PROPERTY = "contact_phone_number"
+HAS_MOBILE_PROPERTY = "has_mobile"
+HAS_MOBILE_PROPERTY_VALUE = "yes"
+CONTACT_PHONE_NUMBER_IS_VERIFIED = "contact_phone_number_is_verified"
+CONTACT_PHONE_NUMBER_IS_VERIFIED_VALUE = 1
+CASE_ITERATION_COUNT = 10000
+MAX_RESCUE_EXCEPTIONS_ON_UPDATE = 5
+CSV_HEADERS = ['Case Id']
+
+TEST_STATES = []
+
+for loc in SQLLocation.active_objects.filter(location_type__code='state', domain=DOMAIN):
+    if loc.metadata.get('is_test_location') == 'test':
+        TEST_STATES.append(loc.name)
+
+
+class Command(BaseCommand):
+    help = """
+    Find the person cases which are
+        - not deleted
+        - not belonging to test locations,
+        - with has_mobile case_property set to "yes",
+            - if the contact_phone_number_is_verified is 1, 
+            update the contact_phone_number property by appending 91 with phone_number,
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.db_alias = None
+        self.log_progress = False
+        self.test_locations = None
+        self.case_accessor = CaseAccessors(DOMAIN)
+
+    def add_arguments(self, parser):
+        parser.add_argument('db_alias')
+        parser.add_argument(
+            '--log',
+            action='store_true',
+            dest='log_progress',
+            default=False,
+            help="log progress"
+        )
+
+    def _case_needs_to_be_updated(self, case):
+        if case.deleted:
+            return False
+        assert case.type == CASE_TYPE
+        if bool(case.owner_id) and case.owner_id in self.test_locations:
+            return False
+        if case.get_case_property(HAS_MOBILE_PROPERTY) == HAS_MOBILE_PROPERTY_VALUE:
+            phone_number = case.get_case_property(PHONE_NUMBER_PROPERTY)
+            phone_number_verified = case.get_case_property(CONTACT_PHONE_NUMBER_IS_VERIFIED)
+            if len(phone_number) == 10 and phone_number_verified == CONTACT_PHONE_NUMBER_IS_VERIFIED_VALUE:
+                return True
+        return False
+
+    def _find_case_ids_with_to_be_updated_phone_number(self):
+        case_ids_with_to_be_updated_phone_number = []
+
+        reindex_accessor = CaseReindexAccessor(
+            domain=DOMAIN,
+            case_type='person', limit_db_aliases=[self.db_alias],
+        )
+
+        filename = 'to_be_updated_phone_numbers_%s_%s.csv' % (self.db_alias, datetime.utcnow())
```

Comment:
the file is a list of case ids that are to be updated, `cases_to_be_updated_for_phone_numbers` or just `cases_to_be_updated` or `case_ids` would be a sufficient suffix to db alias and timestamp to differentiate different file.